### DISCUSSION TITLE
Revealjs slide bullet points break into new lines in a different screen

### DISCUSSION THREAD

**Discussion by zhengchencai** [quarto-dev/quarto-cli#7141] [Q&A] [answered]
### Description

Hello,

I have some bullet points break into new lines in a different screen which has the same resolution (1080p) but just larger size 27inch vs 16inch.

### on 16inch laptop (1080p) 
![image](https://github.com/quarto-dev/quarto-cli/assets/29985544/3fbcbd5d-25bd-4b8b-8f60-ad8e602e764f)
![image](https://github.com/quarto-dev/quarto-cli/assets/29985544/6ac61d97-237d-455d-9c9f-b50f8533d1c7)

### on 27inch monitor (1080p) 
![image](https://github.com/quarto-dev/quarto-cli/assets/29985544/b504df59-c5c6-4b31-b268-52ebaedb156c)
![image](https://github.com/quarto-dev/quarto-cli/assets/29985544/3069be06-bec5-4131-91e8-fd4fe6edcc76)

I think this issue may be related to my SCSS setup, but I'm having trouble resolving it. Could you kindly help me? 

Below, you'll find my code in both qmd and SCSS formats. In essence, I'm trying to ensure that my slides display proportionally on the screen they were designed for (in this case the laptop screen above). In other words, I want the slides to scale in or out, much like a photo, when viewed on different screens.

Thank you very much.

````qmd
---
format: 
  revealjs: 
    theme: theme_test.scss
    embed-resources: true # this controls whether resources in an external folder 
    standalone: true # this is default option to have a complete HTML file 
    transition: none
    progress: true
    controls: false
    width: 1280
    height: 720
    margin: 0
    reference-location: margin
    menu: false
    preview-links: false
    multiplex: false
    auto-stretch: false
    # chalkboard:
    #   theme: whiteboard
    #   boardmarker-width: 5
    #   buttons: false
---

## Test1

This line will turn into a new line on a different screen (same resolution)


- This line will not turn into a new line on a different screen (same resolution)
- This line will turnvv turn into a new line on a different screen (same resolution) resolution
 


## Test2
<br>

<br>


|           _"This line will turn into a new line on a different abcdef screen_ 
|           _This line will turn into a new line on a different abcdef screen_ 
|           _This line will turn into a new line on a different abcdef screen"_ 


<br>

<br>
````

````qmd
/*-- scss:defaults --*/

@import url('https://fonts.googleapis.com/css2?family=Fira+Code:wght@400;700&family=Inter:wght@400;700&family=Patua+One&display=swap');

// fonts
// $body-color: #FFFEFA !default; //dark mode
$font-family-sans-serif: Inter, ui-sans-serif, system-ui, -apple-system, BlinkMacSystemFont, "Segoe UI", Roboto, "Helvetica Neue", Arial, "Noto Sans", sans-serif, "Apple Color Emoji", "Segoe UI Emoji", "Segoe UI Symbol", "Noto Color Emoji" !default;
$font-family-monospace: 'Fira Code', ui-monospace, SFMono-Regular, Menlo, Monaco, Consolas, "Liberation Mono", "Courier New", monospace !default;
$presentation-font-size-root: 26px !default;
$presentation-line-height: 1.3 !default;

// colors
$body-bg: #fff !default;
$body-color: #000 !default;
$link-color: #000 !default;
//$font-weight-bold: 400 !default;
$selection-bg: #FF6720 !default;
//$selection-color: #fff !default;

// headings
$presentation-heading-color: #470A68 !default;
$presentation-heading-font-weight: 600 !default;

/*-- scss:rules --*/

.center-stretch {
  display: flex;
  justify-content: center;
  align-items: center;
  width: 100%;

  figure {
    margin: 0;
    position: relative;
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: center;

    img {
      width: 100%;
      height: auto;
    }

    figcaption {
      position: absolute;
      bottom: 0;
      text-align: left;
      margin-top: 5px; // Adjust the margin as needed
    }
  }
}

.slide-logo {
  display: block !important;
  position: fixed !important;
  top: 0 !important;
  right: 10px !important;
  max-height: 15% !important;
  height: 100% !important;
  width: auto !important;
  color: $body-color !important;
}

.slide-title {
  position: absolute !important;
  top: 60px !important;
  left: 110px !important;
  text-align: left !important;
  font-size: 1.8em !important; 
  color: #470A68;
  padding-right: 10vw !important;
  margin: 0 0 0 0 !important;
}

.slide-title-author {
  position: absolute !important;
  top: 325px !important;
  left: 160px !important;
  text-align: left !important;
  font-size: 1em !important; 
  color: #000;
  margin: 0 0 0 0 !important;
}

.slide-title-email {
  position: absolute !important;
  top: 360px !important;
  left: 160px !important;
  text-align: left !important;
  font-size: 0.9em !important; 
  color: #bfbfbf;
  margin: 0 0 0 0 !important;
}

.slide-title-meeting {
  position: absolute !important;
  top: 420px !important;
  left: 160px !important;
  text-align: left !important;
  font-size: 1em !important; 
  color: #000;
  margin: 0 0 0 0 !important;
}

.slide-title-date {
  position: absolute !important;
  top: 455px !important;
  left: 160px !important;
  text-align: left !important;
  font-size: 0.9em !important; 
  color: #bfbfbf;
  margin: 0 0 0 0 !important;
}

.slide-title-figname {
  position: absolute !important;
  bottom: 112px !important;
  left: 712px !important;
  text-align: left !important;
  font-size: 12px !important; 
  color: #bfbfbf;
  margin: 0 0 0 0 !important;
  white-space: nowrap;
}

.reveal h2 {  
  margin-left: 4vw;
  margin-top: 4.9vh;
  padding-right: 16vw;
  font-size: 1.4em; 
  color: #470A68;
}

.reveal p {
  margin-left: 8vw;
  padding-right: 16vw;
  font-size: 1em;
}

.reveal figure {
  p {
    margin: 0;
    padding: 0;
  }
}

.reveal ul,
.reveal ol,
.reveal li {
  margin-left: 6vw;
  text-indent: 0em; 
  padding-left: 0.5em; 
  padding-right: 6vw; 
  margin-top: 0.1em !important;
  margin-bottom: 0.1em !important;
  font-size: 0.9em; 
}


````

**Comment by cderv** [comment, marked as answer]
I think this is RevealJS re-scaling / zooming feature that is in play here. See in their doc at https://revealjs.com/presentation-size/

You can tweak some config (that you'll find as YAML variable in Quarto https://quarto.org/docs/reference/formats/presentations/revealjs.html#slide-layout

or you can disable completely this feature from revealjs to do your own fixed one.

**Reply by zhengchencai** [reply]
Thanks, @cderv. I experimented with the options you mentioned while adjusting various monitor resolutions. It appears that the issue is indeed related to Reveal.js autoscaling. However, I found the documentation somewhat confusing, particularly in its use of the term "aspect ratio" as mentioned here: ["Reveal will automatically scale presentations uniformly based on the normal size to ensure that everything fits on any given display or viewport without changing the **aspect ratio** or layout of your content"](https://quarto.org/docs/presentations/revealjs/advanced.html). The concept of preserving the same aspect ratio should mean that we scale the slide like a photograph, so that regardless of the screen size and ratio, the content retains its original proportions. However, in practice, it seems that after scaling, the entire slide looks different from the one designed using width and height parameters in the YAML. If there's an image below the text, for instance, it may go off the screen after scaling. 

Do you have any recommendation on how to achieve scaling that uniformly applies a single scaling ratio to everything in the slide? Like when we scale a photo without screwing it. I tried to disable the scale making  ```disable-layout: true```, it will not break the lines for the text, but the slide is still sort of screwed. For instance, the relative postion of the bullet point to the header is different viewed in 27in and 16in screens. 

![image](https://github.com/quarto-dev/quarto-cli/assets/29985544/13f1789a-e6a3-47b5-84cc-a5677742f182)

![image](https://github.com/quarto-dev/quarto-cli/assets/29985544/870dc949-4562-4a6e-a02a-1bcd739c3c23)

Thanks again.

**Reply by zhengchencai** [reply]
I just tried realized it is related to my scss, I disabled the theme and just zoom the slide in browser it is indeed doing a nonscrewed uniform scalling 

![BST_CMRO2_Process1](https://github.com/quarto-dev/quarto-cli/assets/29985544/4ec15129-f3c7-4788-9a44-f67430625f8d)

However, with my scss, it will screw things while zooming.

![BST_CMRO2_Process](https://github.com/quarto-dev/quarto-cli/assets/29985544/ca054d47-e7b8-4c3b-b417-01da4f0d11c7)

Do you know what is wrong with the scss I attached?

**Reply by zhengchencai** [reply]
Sorry for multiple replies. It is indeed the scss issue, I used vw as the unit for margin and padding, if I switch to %, it will behave like a uniform nonscrewed scaling.  Below is when I use % for margin and padding in p and h2, while li is still using vw. You can see p and h2 are scalled uniformaly, but list is scaled to viewpoint I guess since I used vw.

Do you think my understanding is correct? Is % the right unit in my case?


![BST_CMRO2_Process1](https://github.com/quarto-dev/quarto-cli/assets/29985544/66aaf820-fffd-4e29-a6fa-1f87cf7f853b)

**Reply by cderv** [reply]
> Do you think my understanding is correct? Is % the right unit in my case?

Yes I think so. The `vw` unit may not work well with RevealJS zoom & scale feature.
